core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -315,7 +314,7 @@ private[spark] class ExecutorAllocationManager(
   private def onExecutorAdded(executorId: String): Unit = synchronized {
     if (!executorIds.contains(executorId)) {
       executorIds.add(executorId)
-      executorIds.foreach(onExecutorIdle)
+      onExecutorIdle(executorId)
```

Comment:
Hm, but if that's the case we'll never remove the executor in the other cases in `removeExecutor`, i.e.
```
logWarning(s"Attempted to remove unknown executor $executorId!")
```
and
```
logWarning(s"Attempted to remove executor $executorId " +
        s"when it is already pending to be removed!")
```